### PR TITLE
Added PCA prototype on arduino using Lanczos Algorithm

### PR DESCRIPTION
**Pull request by felix642**
Added a prototype to reduce the dimension of a classification dataset. 
This algorithm uses the Lanczos algorithm first and the Jacobi algorithm after.

### CODE REVIEW

**Comment by felix642** [comment]
Also, we need to merge the Jacobi PR before merging this PR

**Review by bisson2000 — APPROVED** [review]
gfm

File: prototype/pca_lanczos/pca.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+def display_results(dataset):
+    np_dataset = np.array(dataset)
+    np_first_dataset = np_dataset[0:49]
+    np_second_dataset = np_dataset[50:99]
+    
+    plt.scatter(np_first_dataset[:, 0], np_first_dataset[:, 1], c="red", label="Iris-setosa")
+    plt.scatter(np_second_dataset[:, 0], np_second_dataset[:, 1], c="blue", label="Iris-versicolor")
+
+    plt.xlabel('Seconde composante principale')
+    plt.ylabel('Première composante principale')
```
Comment (GM07) [inline prototype/pca_lanczos/pca.py:22]: Maybe put the titles in english since the library is in english

**Review by GM07 — COMMENTED** [review]
I left a small comment, but otherwise looks good to me

File: prototype/pca_lanczos/README.md
Comment (nicholaslegrand102) [inline prototype/pca_lanczos/README.md:1]: The README is empty

**Review by nicholaslegrand102 — CHANGES_REQUESTED** [review]
Looks good but it is missing the documentation

**Review by nicholaslegrand102 — APPROVED** [review]
Looks good to me

File: prototype/pca_lanczos/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Principal Component Analysis - Lanczos
+
+## Presentation
+
+This prototype showcase the application of a principal component analysis (PCA) on a classification dataset.
```
Comment (florenceCloutier) [inline prototype/pca_lanczos/README.md:5]: showcases (with an s)

File: prototype/pca_lanczos/pca/pca.ino [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "lanczos.h"
+#include "jacobi.h"
+
+void readElement(void* element, const size_t sizeOfElement) {
```
Comment (florenceCloutier) [inline prototype/pca_lanczos/pca/pca.ino:17]: Include the utils folder to use these functions without code duplication

**Review by florenceCloutier — CHANGES_REQUESTED** [review]
Left a few comments :)

File: prototype/pca_lanczos/pca/pca.ino [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  double mean = 0;
+  for(uint16_t i = 0; i < nbValues; ++i)
+  {
+    mean += ptr[i * stride + offset];
+  }
+  return mean / nbValues;
+}
+
+
+void computeCovarianceMatrix(uint8_t* ptr, uint8_t nbElemsCov, uint8_t nbElems, double* output)
```
Comment (nicholaslegrand102) [inline prototype/pca_lanczos/pca/pca.ino:43]: Please add comments for all of these functions

**Review by nicholaslegrand102 — CHANGES_REQUESTED** [review]
Looks good except for the comments

**Review by florenceCloutier — APPROVED** [review]
Good for me :)

**Review by nicholaslegrand102 — APPROVED** [review]
Looks good to me

**Pull request merged**